common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -281,14 +285,26 @@ def _gated_content_staff_banner(self):
             if content_milestones and self.runtime.user_is_staff:
                 return banner_text
 
+    def _get_course_end(self):
+        """
+        Helper to get course end, because I can't figure out an easier way to do it. Open to suggestions.
+        """
+        current = self
+        while current.location.block_type != 'course':
+            current = current.get_parent()
+            if current == None:
+                raise Exception("Key assumption violated - went past root of course tree without hitting course block")
+        return current.end
```

Comment:
DEBUG: this whole method was written in response to the previous code for getting the course end, `self._get_course().end`, yielding the following error in tests:

```
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/seq_module.py", line 294, in _can_user_view_content
    hidden_date = self._get_course().end if getattr(self, 'self_paced', False) else self.due
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/seq_module.py", line 131, in _get_course
    return self.descriptor.runtime.modulestore.get_course(self.course_id)  # pylint: disable=no-member
  File "/edx/app/edxapp/edx-platform/common/lib/xmodule/xmodule/x_module.py", line 1935, in __getattr__
    return getattr(self._descriptor_system, name)
AttributeError: 'InMemorySystem' object has no attribute 'modulestore'
```